lib/tasks/linkset.rake

```diff
@@ -0,0 +1,23 @@
+namespace :linkset do
+  def find_invalid_link(column_name)
+    Linkset.where.not("#{column_name} ~* ? or #{column_name} = ''",
+      '\Ahttps?:\/\/([^\s:@]+:[^\s:@]*@)?[A-Za-z\d\-]+(\.[A-Za-z\d\-]+)+\.?(:\d{1,5})?([\/?]\S*)?')
+  end
+
+  def clean(links, column_name)
+    links.each do |link|
+      link.update_attribute(column_name.to_sym, nil)
+    end
+  end
+
+  desc "Remove the invalid URLs in linksets"
+  task clean: :environment do
+    links = find_invalid_link("home")
```

Comment:
some points of improvements to consider: use a loop for the columns, use update_all instead of a loop for update the column, use transactions, use a log:

```ruby
%i(home code docs).each do |column|
  Linkset.transaction do
    logger.info "Trying to update #{colunm} values"
    find_invalid_link(table).update_all(["#{column} = ?", nil])
  rescue ...
    # should-never-happen code 
  end
end
```